File: sp/poker/client/src/main/java/com/vpavlov/connection/type/Subtype.java
package com.vpavlov.connection.type;

import java.util.HashMap;
import java.util.Map;

public enum Subtype {
    PING(1)
    ;

    private final static Map<Integer, Subtype> values = new HashMap<Integer, Subtype>();

    static{
        for (Subtype type : Subtype.values()){
            values.put(type.getId(), type);
        }
    }

    private final int id;

    Subtype(int id) {
        this.id = id;
    }

    public int getId() {
        return id;
    }

    public static Subtype getSubtype(int id) {
        return values.get(id);
    }
}

File: sp/poker/server/src/connection/subtype.cpp
#pragma once
#include <map>

class Subtype {
public:
    enum Enum {
        PING = 1
    };

private:
    static const std::map<int, Subtype::Enum> subtypeMap;

public:
    static Enum getSubtype(int id) {
        auto it = subtypeMap.find(id);
        if (it != subtypeMap.end()) {
            return it->second;
        }
        return PING; // Return a default value when the ID is not found.
    }

    static int getId(Enum subtype) {
        for (const auto& entry : subtypeMap) {
            if (entry.second == subtype) {
                return entry.first;
            }
        }
        return -1;  // Handle cases where the subtype ID is not found
    }
};

const std::map<int, Subtype::Enum> Subtype::subtypeMap = {
        {1, Subtype::PING}
};

